diff --git base/BUILD.gn base/BUILD.gn
index f73ed42f84ca486bc0f103370823dbaa14afa35f..f72144503432e2a8bee0ac57fedca4139ba16338 100644
--- base/BUILD.gn
+++ base/BUILD.gn
@@ -39,6 +39,7 @@ import("//build/rust/rust_bindgen.gni")
 import("//build/rust/rust_static_library.gni")
 import("//build/util/process_version.gni")
 import("//build_overrides/build.gni")
+import("//cef/libcef/features/features.gni")
 import("//testing/libfuzzer/fuzzer_test.gni")
 import("//testing/test.gni")
 
@@ -1534,6 +1535,12 @@ component("base") {
     "hash/md5.h",
     "hash/sha1.cc",
     "hash/sha1.h",
+
+  deps += [ "//cef/libcef/features:buildflags" ]
+  if (enable_cef) {
+    configs += [ "//cef/libcef/features:config" ]
+  }
+
   ]
   public_deps += [ "//third_party/boringssl" ]
 
@@ -1950,6 +1957,12 @@ component("base") {
       defines += [ "COM_INIT_CHECK_HOOK_DISABLED" ]
     }
 
+    if (!use_custom_libcxx) {
+      # Enable the VS 2015 Update 2 fix when building with the MSVC standard
+      # library.
+      defines += [ "_ENABLE_ATOMIC_ALIGNMENT_FIX" ]
+    }
+
     libs += [
       "cfgmgr32.lib",
       "ntdll.lib",
diff --git base/allocator/dispatcher/dispatcher.cc base/allocator/dispatcher/dispatcher.cc
index f680f63cffc5b1277e8ce8ce7ef837313907f432..9cb615bbc8a5a3f9eafa477835d6e65538789eb0 100644
--- base/allocator/dispatcher/dispatcher.cc
+++ base/allocator/dispatcher/dispatcher.cc
@@ -8,6 +8,7 @@
 #include "base/check.h"
 #include "base/dcheck_is_on.h"
 #include "base/no_destructor.h"
+#include "cef/libcef/features/features.h"
 #include "partition_alloc/buildflags.h"
 #include "partition_alloc/shim/allocator_shim.h"
 
@@ -33,7 +34,7 @@ struct Dispatcher::Impl {
   }
 
   void Reset() {
-#if DCHECK_IS_ON()
+#if DCHECK_IS_ON() && !BUILDFLAG(IS_CEF_SANDBOX_BUILD)
     DCHECK([&] {
       auto const was_set = is_initialized_check_flag_.test_and_set();
       is_initialized_check_flag_.clear();
diff --git base/process/memory.h base/process/memory.h
index 5d11d4a1560b1275eb4c8360eb4ceb7a2eee9d1b..242a93bcca8ed0b79aa74c475a9e0a1a89a579dd 100644
--- base/process/memory.h
+++ base/process/memory.h
@@ -29,7 +29,7 @@ BASE_EXPORT void EnableTerminationOnOutOfMemory();
 #if PA_BUILDFLAG(USE_PARTITION_ALLOC)
 using partition_alloc::TerminateBecauseOutOfMemory;
 #else
-inline void TerminateBecauseOutOfMemory(size_t) {
+[[noreturn]] inline void TerminateBecauseOutOfMemory(size_t) {
   logging::RawCheckFailure("Out of memory");
 }
 #endif
@@ -58,7 +58,11 @@ bool ReleaseAddressSpaceReservation();
 #if BUILDFLAG(IS_WIN)
 namespace win {
 
+#if PA_BUILDFLAG(USE_PARTITION_ALLOC)
 using partition_alloc::win::kOomExceptionCode;
+#else
+const DWORD kOomExceptionCode = 0xe0000008;
+#endif
 
 }  // namespace win
 #endif
diff --git base/win/sid.cc base/win/sid.cc
index 42f3fc4051a7a33315de243c3a74d377fd47e0ee..b8ca892b14720e4e9d0b44bfabcc86499fa2a613 100644
--- base/win/sid.cc
+++ base/win/sid.cc
@@ -29,12 +29,56 @@
 #include "base/win/scoped_handle.h"
 #include "base/win/scoped_localalloc.h"
 #include "base/win/windows_version.h"
+#include "cef/libcef/features/features.h"
+
+#if !BUILDFLAG(IS_CEF_SANDBOX_BUILD)
 #include "third_party/boringssl/src/include/openssl/sha.h"
+#else
+#include <wincrypt.h>
+#endif
 
 namespace base::win {
 
 namespace {
 
+#if BUILDFLAG(IS_CEF_SANDBOX_BUILD)
+
+#define SHA256_DIGEST_LENGTH 32
+
+bool SHA256(const uint8_t* InData, size_t InDataLen, uint8_t* OutHash) {
+  HCRYPTPROV hProv = 0;
+  HCRYPTHASH hHash = 0;
+
+  if (!CryptAcquireContext(&hProv, nullptr, nullptr, PROV_RSA_AES,
+                           CRYPT_VERIFYCONTEXT)) {
+    return false;
+  }
+
+  if (!CryptCreateHash(hProv, CALG_SHA_256, 0, 0, &hHash)) {
+    CryptReleaseContext(hProv, 0);
+    return false;
+  }
+
+  if (!CryptHashData(hHash, InData, static_cast<DWORD>(InDataLen), 0)) {
+    CryptDestroyHash(hHash);
+    CryptReleaseContext(hProv, 0);
+    return false;
+  }
+
+  DWORD dwHashLen = SHA256_DIGEST_LENGTH;
+  if (!CryptGetHashParam(hHash, HP_HASHVAL, OutHash, &dwHashLen, 0)) {
+    CryptDestroyHash(hHash);
+    CryptReleaseContext(hProv, 0);
+    return false;
+  }
+
+  CryptDestroyHash(hHash);
+  CryptReleaseContext(hProv, 0);
+  return true;
+}
+
+#endif  // BUILDFLAG(IS_CEF_SANDBOX_BUILD)
+
 template <typename Iterator>
 Sid FromSubAuthorities(const SID_IDENTIFIER_AUTHORITY& identifier_authority,
                        size_t sub_authority_count,
diff --git build_overrides/dawn.gni build_overrides/dawn.gni
index cec3df3e50b6ecec39dc8be1a6efd9f2a7403b20..309b4e6a4fe0d7b6cbec1e695042b9d5dab8ddc0 100644
--- build_overrides/dawn.gni
+++ build_overrides/dawn.gni
@@ -2,11 +2,15 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+import("//base/allocator/partition_allocator/partition_alloc.gni")
+
 # The paths to Dawn's dependencies
 dawn_angle_dir = "//third_party/angle"
 dawn_glfw_dir = "//third_party/dawn/third_party/glfw"
 dawn_googletest_dir = "//third_party/googletest/src"
-dawn_partition_alloc_dir = "//base/allocator/partition_allocator"
+if (use_partition_alloc) {
+  dawn_partition_alloc_dir = "//base/allocator/partition_allocator"
+}
 dawn_jinja2_dir = "//third_party/jinja2"
 dawn_jsoncpp_dir = "//third_party/jsoncpp"
 dawn_spirv_tools_dir = "//third_party/spirv-tools/src"
