diff --git base/logging.cc base/logging.cc
index 3a470897fa..44ba822e1b 100644
--- base/logging.cc
+++ base/logging.cc
@@ -49,6 +49,7 @@
 #include "base/threading/platform_thread.h"
 #include "base/trace_event/interned_args_helper.h"
 #include "base/trace_event/typed_macros.h"
+#include "cef/libcef/features/features.h"
 #include "base/vlog.h"
 #include "build/build_config.h"
 #include "third_party/abseil-cpp/absl/base/internal/raw_logging.h"
@@ -119,7 +120,7 @@ typedef FILE* FileHandle;
 #include "base/fuchsia/scoped_fx_logger.h"
 #endif
 
-#if !BUILDFLAG(IS_NACL)
+#if !BUILDFLAG(IS_NACL) && !BUILDFLAG(IS_CEF_SANDBOX_BUILD)
 #include "base/logging/rust_logger.rs.h"
 #endif
 
@@ -442,7 +443,30 @@ void WriteToFd(int fd, const char* data, size_t length) {
   }
 }
 
+#if BUILDFLAG(IS_WIN)
+using SetLogFatalCrashKeyFunc = void (*)(const char* /*file*/,
+                                         int /*line*/,
+                                         const char* /*message*/);
+
+SetLogFatalCrashKeyFunc SetLogFatalCrashKeyFuncGetter() {
+  static SetLogFatalCrashKeyFunc log_fatal_crash_key_func = []() {
+    // Function exported by bootstrap.exe.
+    return reinterpret_cast<SetLogFatalCrashKeyFunc>(
+        GetProcAddress(GetModuleHandle(NULL), "SetLogFatalCrashKey"));
+  }();
+  return log_fatal_crash_key_func;
+}
+#endif  // BUILDFLAG(IS_WIN)
+
 void SetLogFatalCrashKey(LogMessage* log_message) {
+#if BUILDFLAG(IS_WIN)
+  if (auto func = SetLogFatalCrashKeyFuncGetter()) {
+    func(log_message->file(), log_message->line(),
+         log_message->str().c_str() + log_message->message_start());
+    return;
+  }
+#endif
+
 #if !BUILDFLAG(IS_NACL)
   // In case of an out-of-memory condition, this code could be reentered when
   // constructing and storing the key. Using a static is not thread-safe, but if
@@ -521,6 +545,11 @@ bool BaseInitLoggingImpl(const LoggingSettings& settings) {
   g_log_format = settings.log_format;
 #endif
 
+#if BUILDFLAG(IS_WIN)
+  // Preload the function pointer so that we do minimal work while crashing.
+  SetLogFatalCrashKeyFuncGetter();
+#endif
+
   MaybeInitializeVlogInfo();
 
   g_logging_destination = settings.logging_dest;
@@ -531,7 +560,7 @@ bool BaseInitLoggingImpl(const LoggingSettings& settings) {
   }
 #endif
 
-#if !BUILDFLAG(IS_NACL)
+#if !BUILDFLAG(IS_NACL) && !BUILDFLAG(IS_CEF_SANDBOX_BUILD)
   // Connects Rust logging with the //base logging functionality.
   internal::init_rust_log_crate();
 #endif
diff --git base/logging.h base/logging.h
index 262a86ef023b2871b2ec94e4d42fb93480a17cc1..2a638248f5da3fb23d9c8cf5596562671edb15b8 100644
--- base/logging.h
+++ base/logging.h
@@ -613,6 +613,7 @@ class BASE_EXPORT LogMessage {
   std::string str() const { return stream_.str(); }
   const char* file() const { return file_; }
   int line() const { return line_; }
+  size_t message_start() const { return message_start_; }
 
   // Gets file:line: message in a format suitable for crash reporting.
   std::string BuildCrashString() const;
